diff --git a/src/bindings/python/pycsp.c b/src/bindings/python/pycsp.c
index efff44e..34bab3e 100644
--- a/src/bindings/python/pycsp.c
+++ b/src/bindings/python/pycsp.c
@@ -6,6 +6,7 @@
 #include <csp/csp_cmp.h>
 #include <csp/interfaces/csp_if_zmqhub.h>
 #include <csp/interfaces/csp_if_kiss.h>
+#include <csp/interfaces/csp_if_udp.h>
 #include <csp/drivers/usart.h>
 #include <csp/drivers/can_socketcan.h>
 #include <endian.h>
@@ -890,6 +891,39 @@ static PyObject * pycsp_kiss_init(PyObject * self, PyObject * args) {
 	Py_RETURN_NONE;
 }
 
+static PyObject * pycsp_udp_init(PyObject * self, PyObject * args) {
+	uint16_t addr;
+	char * host;
+	int lport;
+	int rport;
+	int is_default = 0;
+	uint16_t mask = 8;
+
+	if (!PyArg_ParseTuple(args, "Hsii|pH", &addr, &host, &lport, &rport,
+	                      &is_default, &mask)) {
+		return NULL;
+	}
+
+	static csp_iface_t        iface;
+	static csp_if_udp_conf_t  udp_conf;
+	static char               udp_host[64];
+
+	strncpy(udp_host, host, sizeof(udp_host) - 1);
+	udp_host[sizeof(udp_host) - 1] = '\0';
+
+	udp_conf.host  = udp_host;
+	udp_conf.lport = lport;
+	udp_conf.rport = rport;
+
+	csp_if_udp_init(&iface, &udp_conf);
+
+	iface.addr       = addr;
+	iface.netmask    = mask;
+	iface.is_default = is_default;
+
+	Py_RETURN_NONE;
+}
+
 static PyObject * pycsp_packet_set_data(PyObject * self, PyObject * args) {
 	PyObject * packet_capsule;
 	Py_buffer data;
@@ -997,6 +1031,7 @@ static PyMethodDef methods[] = {
 	{"zmqhub_init", pycsp_zmqhub_init, METH_VARARGS, ""},
 #endif /* CSP_HAVE_LIBZMQ */
 	{"kiss_init", pycsp_kiss_init, METH_VARARGS, ""},
+	{"udp_init",  pycsp_udp_init,  METH_VARARGS, ""},
 
 	/* csp/drivers/can_socketcan.h */
 	{"can_socketcan_init", pycsp_can_socketcan_init, METH_VARARGS, ""},
